Added real type

juliaofferman opened the pull request.

juliaofferman committed:
Add list generator to TestGen.ml

juliaofferman committed:
Add rev and list equality components

juliaofferman committed:
Add real number type

juliaofferman committed:
Add real number value

juliaofferman committed:
Create RealComponents.ml

juliaofferman committed:
change can_apply to is_argument_valid

juliaofferman committed:
Update RealComponents.ml

juliaofferman committed:
Update RealComponents.ml

juliaofferman committed:
Update RealComponents.ml

juliaofferman committed:
Update RealComponents.ml

juliaofferman committed:
Update RealComponents.ml

juliaofferman committed:
Add Real components

juliaofferman committed:
Add real type

juliaofferman committed:
Additional job examples

juliaofferman committed:
Update TestGen.ml

juliaofferman committed:
Update TestGen.ml

juliaofferman committed:
Update real constants

juliaofferman committed:
Update Real component to_string's

juliaofferman committed:
Update TestGen.ml

juliaofferman committed:
Update App.ml

juliaofferman committed:
Update App.ml

juliaofferman committed:
Tests for real components

juliaofferman committed:
Update to include real component tests

juliaofferman committed:
Update Test_RealComponents.ml

juliaofferman committed:
Update Test_RealComponents.ml

juliaofferman committed:
Update Test_RealComponents.ml

juliaofferman committed:
Add additional tests

juliaofferman committed:
Fix negation

SaswatPadhi commented:
Thanks for working on this PR, Julia!

I finally have everything setup and will be reviewing this soon.

juliaofferman committed:
Adding "LRA" logic

juliaofferman committed:
Remove scaling from levels

juliaofferman committed:
Various changes to support Real type

Added case for evaluating sexp's using division, updated of_atomic_string to support reals

juliaofferman committed:
Add linear multiplication and division

juliaofferman committed:
Remove unnecessary log

juliaofferman committed:
Fix spacing

juliaofferman committed:
Add real tests

juliaofferman committed:
LRA benchmark

juliaofferman committed:
Add real benchmarks

juliaofferman committed:
Removie real ite component

Removie Real ite component and unnecessary of_string function
